src/filesystem/implementation.js

```diff
@@ -279,12 +279,52 @@ function find_node(context, path, callback) {
 }
 
 /**
- * find_lnode
+ * find_symlink_node
  */
 // in: file or directory path
-// out: node structure, or error
-function find_symbolic_node(context, path, callback) {
-  lstat_file(context, path, callback);
+// out: symlink node structure, or error
+function find_symlink_node(context, path, callback) {
+  path = normalize(path);
+  var name = basename(path);
+  var parentPath = dirname(path);
+
+  var directoryNode;
+  var directoryData;
+
+  if(ROOT_DIRECTORY_NAME === name) {
+    find_node(context, path, callback);
+  } else {
+    find_node(context, parentPath, read_directory_data);
```

Comment:
Question: won't this just undo your work here, and give back a de-referenced symlink via `find_node`?  I thought the goal here was to get back the symlink vs. linked file node.

Reply:
That seems to make sense and I missed that, but lstat_file seems to be using this approach?

Reply:
What is the difference between stat and node really?

Reply:
`find_node` and `stat` are basically identical: they both get a File Node object from the database for a given path.

I think I'm wrong, looking at `lstat_file`:  https://github.com/filerjs/filer/blob/master/src/filesystem/implementation.js#L897-L939.  It's only getting the base dir, and then using that to read out the directory entries.  A directory node's data is just an array of filenames stored in the dir.

It's using `find_node` just like this.  I assume you've pulled your code from this?  If so, can you also change `lstat_file` to just call your new function?